lib/ae_mdw/db/sync/contract.ex

```diff
@@ -180,4 +184,31 @@ defmodule AeMdw.Db.Sync.Contract do
         |> NameRevokeMutation.new(call_txi, block_index)
     end)
   end
+
+  # Temporary HACK to fix oracle query nonces.
+  defp fix_oracle_queries(events, block_hash) do
+    {fixed_events, _acc} =
+      Enum.map_reduce(events, %{}, fn
+        {"Oracle.query", aetx}, accounts_nonces ->
+          {:oracle_query_tx, tx} = :aetx.specialize_type(aetx)
+          sender_pk = :aeo_query_tx.sender_pubkey(tx)
+
+          nonce =
+            case Map.fetch(accounts_nonces, sender_pk) do
+              {:ok, nonce} -> nonce
+              :error -> Db.nonce_at_block(block_hash, sender_pk)
```

Comment:
The nonces map is part of the events reduce call, so it would only be called once per sender_pk. 

In addition, oracle query calls rarely ever happen, so in reality it would rarely ever be called. 

Also, once the fix from AE is in place for oracle query nonces, we will get rid of this.

Reply:
The account tree is per block_hash, could be reused for multiple senders. I cannot predict how it gonna be used with Hyperchains but it's okay for now.